### ISSUE TITLE
GameLevel interface data structure

### ISSUE THREAD

**Issue by stevenkhong** [MrFool/Cattac#38]
**Current Implementation**
Array of string, indicating doodad name, or blank if no doodad/wall, implemented in a dictionary as such

```
[
    1: "",
    2: "wall",
    ...
]
```

Problems: limited to node, does not indicate other game level properties, inefficient empty nodes being sent around

**Proposed Implementation**
This implementation allows for more flexible transfer of data. Main driving force for this change is that now _wormhole is broken_! :(

```
[
    "rows": 10,
    "cols": 10,
    "type": "basic"
    "entities": [
        [
            "row": 1,
            "col": 6,
            "name": wall
        ],
        [
            "row": 5,
            "col": 5,
            "name": wormhole,
            "destNodeRow": 8,
            "destNodeCol": 1
        ],
        ...
    ]
]
```

Refer to commit below for my shameful half done work.

**Comment by junhaoyap** [comment]
Looks good!

**Comment by junhaoyap** [comment]
Didn't notice you're already on it, keep at it, I'll take up another issue instead

**Comment by stevenkhong** [comment]
First working version
[branch](https://github.com/MrFool/Cattac/tree/improvement/gamelevel-data-structure-redesign)

Need help with 4p testing.

**Comment by stevenkhong** [comment]
@MrFool you might want to merge this branch before carrying on with Connection Manager #31

**Comment by junhaoyap** [comment]
No problem, I'll check out the branch and test this along with wudi tomorrow and fix it to make it work (or maybe if it works right out of the box, then no changes needed xD)

**Comment by stevenkhong** [comment]
GameLevel Structure:

```
[
    "rows": 10,
    "cols": 10,
    "type": "basic"
    "entities": [
        [
            "row": 1,
            "col": 6,
            "name": wall
        ],
        [
            "row": 5,
            "col": 5,
            "name": wormhole,
            "destNode": 
                [
                    "row": 4,
                    "col": 1
                ]
        ],
        ...
    ]
]
```

**Comment by stevenkhong** [comment]
Merged #76
